src/lib/badge/badge.spec.ts

```diff
@@ -0,0 +1,93 @@
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {Component, DebugElement} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {MatBadge, MatBadgeModule} from './index';
+import {ThemePalette} from '@angular/material/core';
+
+describe('MatBadge', () => {
+  let fixture: ComponentFixture<any>;
+  let testComponent: BadgeWithTextContent;
+  let badgeNativeElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [MatBadgeModule],
+      declarations: [BadgeWithTextContent],
+    });
+
+    TestBed.compileComponents();
+    fixture = TestBed.createComponent(BadgeWithTextContent);
+    testComponent = fixture.debugElement.componentInstance;
+    fixture.detectChanges();
+  });
+
+  describe('MatBadge Text', () => {
+    let badgeDebugElement: DebugElement;
+
+    beforeEach(() => {
+      badgeDebugElement = fixture.debugElement.query(By.directive(MatBadge));
+      badgeNativeElement = badgeDebugElement.nativeElement;
+      fixture.detectChanges();
+    });
+
+    it('should update the badge based on attribute', () => {
+      let badgeContentDebugElement = badgeNativeElement.querySelector('.mat-badge-content');
+
+      expect(badgeContentDebugElement.textContent).toContain('1');
+
+      testComponent.badgeContent = '22';
+      fixture.detectChanges();
+
+      badgeContentDebugElement = badgeNativeElement.querySelector('.mat-badge-content');
+      expect(badgeContentDebugElement.textContent).toContain('22');
+    });
+
+    it('should apply class based on color attribute', () => {
+      testComponent.badgeColor = 'primary';
+      fixture.detectChanges();
+      expect(badgeNativeElement.classList.contains('mat-badge-primary')).toBe(true);
+
+      testComponent.badgeColor = 'accent';
+      fixture.detectChanges();
+      expect(badgeNativeElement.classList.contains('mat-badge-accent')).toBe(true);
+
+      testComponent.badgeColor = 'warn';
+      fixture.detectChanges();
+      expect(badgeNativeElement.classList.contains('mat-badge-warn')).toBe(true);
+
+      testComponent.badgeColor = null;
+      fixture.detectChanges();
+
+      expect(badgeNativeElement.classList).not.toContain('mat-badge-accent');
+    });
+
+    it('should update the badget position on direction change', () => {
+      expect(badgeNativeElement.classList.contains('mat-badge-above')).toBe(true);
+      expect(badgeNativeElement.classList.contains('mat-badge-after')).toBe(true);
+
+      testComponent.badgeDirection = 'below before';
+      fixture.detectChanges();
+
+      expect(badgeNativeElement.classList.contains('mat-badge-below')).toBe(true);
+      expect(badgeNativeElement.classList.contains('mat-badge-before')).toBe(true);
+    });
+  });
```

Comment:
There should also be tests for the description, overlay, size, and hidden properties.